runtime/drivers/duckdb/motherduck/motherduck.go

```diff
@@ -0,0 +1,42 @@
+package motherduck
+
+import (
+	"context"
+
+	"github.com/rilldata/rill/runtime/drivers"
+)
+
+func init() {
+	drivers.RegisterConnector("motherduck", connector{})
+}
+
+var spec = drivers.Spec{
+	DisplayName: "MotherDuck",
+	Description: "Import data from MotherDuck.",
+	Properties: []drivers.PropertySchema{
+		{
+			Key:         "query",
+			Type:        drivers.StringPropertyType,
+			Required:    true,
+			DisplayName: "Query",
+			Description: "Query to extract data from MotherDuck.",
+			Placeholder: "select * from my_db.my_table;",
+		},
+	},
+	ConnectorVariables: []drivers.VariableSchema{
+		{
+			Key:    "token",
+			Secret: true,
+		},
+	},
+}
+
+type connector struct{}
+
+func (c connector) Spec() drivers.Spec {
+	return spec
+}
+
+func (c connector) HasAnonymousAccess(ctx context.Context, props map[string]any) (bool, error) {
+	return false, nil
+}
```

Comment:
I think what throws me off is the implicit coupling between the connector name and the driver name, and the implicit relationship between the connector spec and the driver config properties.

I'm not really able to put into words what the difference is between a "connector" and a "driver" anymore, so that's why I thought we should unify.

Any other ideas for how we can reduce the confusion?